### PR TITLE
added fix for WorkUCC

### PR DESCRIPTION
**Pull request by Fuzzkatt**
Added new constructor for WorkUCC to take in optional inputTensors argument for to enable record_shapes=True for profiling purposes. Tested at https://github.com/pytorch/pytorch/pull/84323 which manually merges in https://github.com/pytorch/pytorch/pull/83285.

### CODE REVIEW

File: torch/csrc/distributed/c10d/ProcessGroupUCC.hpp [hunk 1]
```diff
@@ -165,6 +165,12 @@ class TORCH_API ProcessGroupUCC : public ProcessGroup {
         const char* prof_title,
         const c10::intrusive_ptr<ProcessGroupUCCLogger>& logger)
         : ProcessGroup::Work(-1, opType, prof_title), logger_(logger) {}
+    WorkUCC(
```
Comment (zasdfgbnm) [inline torch/csrc/distributed/c10d/ProcessGroupUCC.hpp:168]: Do we still need the constructor that doesn't have inputs?

Comment (kumpera) [reply torch/csrc/distributed/c10d/ProcessGroupUCC.hpp:168]: There's one use left of the old ctor but it looks like it should use the new one.

**Review by kumpera — DISMISSED** [review]

File: torch/csrc/distributed/c10d/ProcessGroupUCC.hpp (hunk 1 above)
Comment (Fuzzkatt) [reply torch/csrc/distributed/c10d/ProcessGroupUCC.hpp:161]: Removed old constructor. Couldn't find the one use left of the old constructor that @kumpera mentioned.

**Comment by zasdfgbnm** [comment]
cc: @kingchc

**Review by kingchc — APPROVED** [review]
Looks good to me! Please take care of linter issues.

**Comment by zasdfgbnm** [comment]
@Fuzzkatt Could you rebase this PR to see if the failure goes away?

**Comment by zasdfgbnm** [comment]
@kumpera @kwen2501 Could you take a look at this PR? The failing workflow seems unrelated, and this should ready to go.

**Review by kwen2501 — APPROVED** [review]
LGTM

**Pull request closed without merging**